src/Datadog.Trace.ClrProfiler.Managed/Integrations/Testing/NUnitIntegration.cs

```diff
@@ -0,0 +1,358 @@
+            TargetMaximumVersion = Major3,
+            TargetSignatureTypes = new[] { NUnitTestResultType, NUnitTestExecutionContextType })]
+        public static object TestCommand_Execute(
+            object testMethodCommand,
+            object testExecutionContext,
+            int opCode,
+            int mdToken,
+            long moduleVersionPtr)
+        {
+            if (testMethodCommand == null) { throw new ArgumentNullException(nameof(testMethodCommand)); }
+
+            Type testMethodCommandType = testMethodCommand.GetType();
+            Func<object, object, object> execute;
+
+            try
+            {
+                execute = MethodBuilder<Func<object, object, object>>
+                    .Start(moduleVersionPtr, mdToken, opCode, NUnitExecuteMethod)
+                    .WithConcreteType(testMethodCommandType)
+                    .WithParameters(testExecutionContext)
+                    .WithNamespaceAndNameFilters(NUnitTestResultType, NUnitTestExecutionContextType)
+                    .Build();
+            }
+            catch (Exception ex)
+            {
+                Log.ErrorRetrievingMethod(
+                    exception: ex,
+                    moduleVersionPointer: moduleVersionPtr,
+                    mdToken: mdToken,
+                    opCode: opCode,
+                    instrumentedType: NUnitTestCommandType,
+                    methodName: NUnitExecuteMethod,
+                    instanceType: testMethodCommandType.AssemblyQualifiedName);
+                throw;
+            }
+
+            if (testMethodCommandType.FullName != NUnitTestMethodCommandType &&
+                testMethodCommandType.FullName != NUnitSkipCommandType)
+            {
+                return execute(testMethodCommand, testExecutionContext);
+            }
+
+            Scope scope = CreateScope(testExecutionContext, testMethodCommandType);
+            if (scope is null)
+            {
+                return execute(testMethodCommand, testExecutionContext);
+            }
+
+            using (scope)
+            {
+                object result = null;
+                Exception exception = null;
+                try
+                {
+                    scope.Span.ResetStartTime();
+                    result = execute(testMethodCommand, testExecutionContext);
+                }
+                catch (Exception ex)
+                {
+                    exception = ex;
+                    throw;
+                }
+                finally
+                {
+                    FinishScope(scope, testExecutionContext, exception);
+                }
+
+                return result;
+            }
+        }
+
+        private static Scope CreateScope(object testExecutionContext, Type testMethodCommandType)
+        {
+            Scope scope = null;
+
+            try
+            {
+                if (testExecutionContext.TryGetPropertyValue<object>("TestObject", out object testObject) &&
```

Comment:
Is there a reason we need to extract the testObject? It doesn't look like the result is used

Reply:
fixed in: https://github.com/DataDog/dd-trace-dotnet/pull/870/commits/af50684975ef32b3416e4f8567fc0eb48ec5e841